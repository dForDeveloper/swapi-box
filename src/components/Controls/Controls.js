import React from 'react';
import PropTypes from 'prop-types';

export const Controls = ({ favCount, activeCategory, setActiveCategory }) => {
  const buttonNames = ['people', 'planets', 'vehicles', 'favorites'];
  const buttons = buttonNames.map(name => {
    const buttonClass = name === activeCategory ? 'button-active' : '';
    const buttonText = name === 'favorites' ? `favorites ${favCount}` : name;
    return (
      <button
        className={buttonClass}
        key={name}
        onClick={() => setActiveCategory(name)}
      >
        {buttonText}
      </button>
    );
  });
  return <nav>{buttons}</nav>;
}

Controls.propTypes = {
  favCount: PropTypes.number,
  activeCategory: PropTypes.string,
  setActiveCategory: PropTypes.func
}