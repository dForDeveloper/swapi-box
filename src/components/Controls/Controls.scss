.Controls {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  justify-content: space-around;
  margin: 15px 0;

  .button {
    width: 50%;
    min-width: 140px;
    background-color: black;
    border: 3px solid white;
    color: white;
    height: 50px;
    border-radius: 25px;
    margin: auto;
    text-transform: uppercase;
    outline: none;
    font-size: 1.2rem;
  }
  
  .button--active {
    @extend .button;
    // background-color: black;
    border: 3px solid yellow;
    color: yellow;
  }
}
