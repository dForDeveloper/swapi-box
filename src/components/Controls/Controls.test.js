import React from 'react';
import { Controls } from './Controls';
import { shallow } from 'enzyme';

describe('Controls', () => {
  it('should match the snapshot', () => {
    const mockSetActiveCategory = jest.fn();
    let wrapper = shallow(
      <Controls
        favCount={0}
        activeCategory={''}
        setActiveCategory={mockSetActiveCategory}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});