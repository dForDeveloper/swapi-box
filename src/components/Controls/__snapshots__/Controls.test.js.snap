// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
<nav
  className="Controls"
>
  <button
    className="button"
    key="people"
    onClick={[Function]}
  >
    people
  </button>
  <button
    className="button"
    key="planets"
    onClick={[Function]}
  >
    planets
  </button>
  <button
    className="button"
    key="vehicles"
    onClick={[Function]}
  >
    vehicles
  </button>
  <button
    className="button"
    key="favorites"
    onClick={[Function]}
  >
    favorites 0
  </button>
</nav>
`;
